core/src/main/java/org/elasticsearch/search/aggregations/AggregatorParsers.java

```diff
@@ -60,10 +60,12 @@ public AggregatorParsers(Set<Aggregator.Parser> aggParsers, Set<PipelineAggregat
         Map<String, Aggregator.Parser> aggParsersBuilder = new HashMap<>(aggParsers.size());
         for (Aggregator.Parser parser : aggParsers) {
             aggParsersBuilder.put(parser.type(), parser);
-            AggregatorFactory factoryPrototype = parser.getFactoryPrototype();
+            AggregatorFactory[] factoryPrototypes = parser.getFactoryPrototypes();
             // NORELEASE remove this check when agg refactoring complete
-            if (factoryPrototype != null) {
-                namedWriteableRegistry.registerPrototype(AggregatorFactory.class, factoryPrototype);
+            if (factoryPrototypes != null) {
```

Comment:
if we return an array, I'd rather like to enforce it to be non-null: if you don't want to register something you can provide an empty array?